File: Source/Bifrost.JavaScript.Specs/utils/for_StringMapping/when_matching_a_string_that_should_not_match.js
describe("when matching a string that should not match", function () {
    var mapping = Bifrost.utils.StringMapping.create({
        format: "{something}/{else}",
        mappedFormat: "whatevva"
    });
    var result = mapping.matches("hello");

    it("should not match", function () {
        expect(result).toBe(false);
    });
});

File: Source/Bifrost.Specs/Utils/for_StringMapping/when_matching_a_string_that_should_not_match.cs
using Bifrost.Utils;
using Machine.Specifications;

namespace Bifrost.Specs.Utils.for_StringMapping
{
    public class when_matching_a_string_that_should_not_match
    {
        static StringMapping mapping = new StringMapping(
                "{something}/{else}",
                "whatevva"
            );
        static bool result;

        Because of = () => result = mapping.Matches("hello");

        It should_not_match = () => result.ShouldBeFalse();
    }
}
